src/wms/src/main/java/org/geoserver/wms/decoration/LegendDecoration.java

```diff
@@ -101,6 +107,22 @@ public Dimension findOptimalSize(Graphics2D g2d, WMSMapContent mapContext) {
                 request.setKvp(dispatcherRequest.getKvp());
                 request.setRawKvp(dispatcherRequest.getRawKvp());
             }
+            // online resource handling
+            LayerInfo info = wms.getLayerByName(layer.getTitle());
```

Comment:
All this preparation is done to get the LegendInfo, right? If so, best to move it in a separate method, and just return the legend info, without cluttering this method with otherwise unused variables.

Reply:
applied